public/app/features/alerting/unified/components/rule-editor/alert-rule-form/simplifiedRouting/AlertManagerRouting.tsx

```diff
@@ -18,15 +18,45 @@ interface AlertManagerManualRoutingProps {
   alertManager: AlertManagerDataSource;
 }
 
+const LOADING_SPINNER_DURATION = 1000;
+
 export function AlertManagerManualRouting({ alertManager }: AlertManagerManualRoutingProps) {
   const styles = useStyles2(getStyles);
 
   const alertManagerName = alertManager.name;
-  const { isLoading, error: errorInContactPointStatus, contactPoints } = useContactPointsWithStatus();
+  const { isLoading, error: errorInContactPointStatus, contactPoints, refetchReceivers } = useContactPointsWithStatus();
   const [selectedContactPointWithMetadata, setSelectedContactPointWithMetadata] = useState<
     ContactPointWithMetadata | undefined
   >();
 
+  const [loadingContactPoints, setLoadingContactPoints] = useState(false);
+
+  // we need to keep track of the timeout id, so we can clear it when the user clicks the refresh button, or when the component unmounts
+  const [timeOutId, setTimeOutId] = useState<NodeJS.Timeout | null>(null);
+
+  const onClickRefresh = () => {
+    // we need to clear the timeout when the user clicks the refresh button, as are creating a new timeout
+    if (timeOutId) {
+      clearTimeout(timeOutId);
+    }
+    refetchReceivers();
```

Comment:
This one probably needs a `.finally()` because otherwise there would be a race condition with `refetchReceivers()` if it that request takes longer than 1s.

In that scenario the UI would stop showing the spinner whilst we're actually still fetching  :)

Reply:
I've just pushed a commit to cover this use case:

https://github.com/grafana/grafana/assets/33540275/52a2e9ba-ba2e-4779-81e6-f4c0b60025f5